Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -420,7 +420,12 @@ class BrowserViewController: UIViewController,
             urlBar.locationView.tabDidChangeContentBlocking(tab)
         }
 
-        _ = tabManager.startAtHomeCheck()
+        if tabManager.startAtHomeCheck() {
+            guard let presentedViewController else { return }
+            if presentedViewController.modalPresentationStyle == .popover {
+                dismissVC()
+            }
+        }
```

Comment:
Is there a way to move this logic into the coordinator when the new tab is opened?

cc @lmarceau

Reply:
Mh. Just realized for the URL bar CFR we hold off presenting it until the start at home check is done.

```
        // Checks if the app started with the Homepage tab
        // If true, don't show the CFR to avoid showing it on the Homepage
        guard !didStartAtHome else {
            didStartAtHome = false
            return
        }
```

So if we `didStartAtHome = tabManager.startAtHomeCheck()` instead it will solve it as this check protects the shopping CFR 🤔  
```        
let contextualViewProvider = ContextualHintViewProvider(forHintType: .shoppingExperience,
                                                                                               with: profile)
```

Why I say this is because CFRs aren't part of coordinator work as of now, and this change seems minimal enough if it works?

Reply:
If we use didStartAtHome = tabManager.startAtHomeCheck() and it returns true, we need to immediately set didStartAtHome to false. Otherwise, if there is a tab open, the shopping CFR won't show, only if we refresh the page. This PR checks in case the CFR was not closed. The user has put the app in the background for over 4 hours and then put it in the foreground, a new homepage tab is created, and we need to check if the CFR is on the screen and dismiss it; otherwise, it will be left hanging on the homepage tab.

This check is when we want to present the CFR and we cold start the app:

// Checks if the app started with the Homepage tab
// If true, don't show the CFR to avoid displaying it on the Homepage
guard !didStartAtHome else {
    didStartAtHome = false
    return
}

This check is when the CFR was not closed:

if tabManager.startAtHomeCheck() {
    guard let presentedViewController else { return }
    if presentedViewController.modalPresentationStyle == .popover {
        dismissVC()
    }
}

didStartAtHome = tabManager.startAtHomeCheck() won't fix the problem because the CFR is already on the screen.

We could also do this, but I don't see the point in doing it, instead we could check directly, as in the PR.

didStartAtHome = tabManager.startAtHomeCheck()
if didStartAtHome {
    didStartAtHome = false
    guard let presentedViewController else { return }
    if presentedViewController.modalPresentationStyle == .popover {
        dismissVC()
    }
}

Reply:
I guess we'll go the coordinator route then. I have [a branch](https://github.com/mozilla-mobile/firefox-ios/compare/main...lmarceau:firefox-ios:lm/test-for-fakespot-cfr) that solve the problem I believe, let me know if that works.

Reply:
It works well, TY! Good Job!!